Spy: Add new item spy knife

Xackii opened the pull request:
## About The Pull Request
Video:
https://youtu.be/X4ajeO59Szw
Text:
Adds a new item to the spy bounty - spy knife:
	force 10
	throwforce 20
	armour penetration 50

Additional abilities

 Attack in the back deals 30 additional damage.

 Attacking from a prone position will cause the target to be knocked down for 4 seconds.

 Attacks against a prone target deal 15 additional damage

All additional abilities only work if the knife is used by a spy.

## Why It's Good For The Game

Gives something unique to the spy to differentiate him from other antagonists(and graytiders)

:cl:
add: New spy knife in spy bounty
/:cl:

Xackii committed:
spyknife

Rhials commented:
I love the item itself, but this really shouldn't be a roundstart item for spies. I'm pretty sure there's a reason we've moved away from spawning players with "this user is an antagonist" gear in their bag (codex cicatrix, supermatter sliver kit, etc)

Pickle-Coding commented:
Should only penetrate armour if the armour isn't airtight or whatever. Wouldn't make sense if some random knife can pierce through an adminium modsuit or something. Also it's hard to tell what direction a prone target is facing.

Pickle-Coding commented:
Also I don't see any way how someone would figure out how to use this in game.

Likteer reviewed (approved):
Baguette Hon Hon Hon

Dawnseer commented:
Why not just replace the knife you can get as a bounty with this? Instead of giving all 7-10 spies of a round this?

necromanceranne commented:
> Should only penetrate armour if the armour isn't airtight or whatever. Wouldn't make sense if some random knife can pierce through an adminium modsuit or something.

I dont think introducing another layer to an armor system that is already difficult for a player to parse in-game is going to do it much favours. What even constitutes 'airtight' is somewhat arbitrary in this game, and would be tedious to ensure it is applied properly in every possible location that it could show up. Something could be 'airtight' but almost certainly not realistcally stop a knife puncture, like plasmamen jumpsuits, which are literally described as 'airtight'.

Without fundamentally changing the armor system to something more simple overall, I think it would be best to not complicate it further than needed, as that only introduces more oversights as content is added for not much gain.

> Also why would lying down let you knock down an opponent?

You stab them in the thigh according to the code. It isn't conveyed very clearly and definitely needs some kind of chat message/popup, but I imagine that's what the 'thigh_hit' bit is about.

Helg2 commented:
London RP

Fikou commented:
backstabs in ss13 are stupid it is not a first person shooter, you have 360 vision and tg does not have any facing/strafing mechanics people run front and back turning 180 degrees in combat all the time

Higgin commented:
everything about this is cracked. 2-tap crit from backstab, knockdown from prone, extra damage (just shy of an e-sword) on prone? built-in snowflake antag check? (lay down and stab this monkey for me please)

maybe as like a lategame high-risk bounty, I guess, but this broadly seems terribly imbalanced and dominant enough on its own in the right hands to not put within a mile of roundstart. this is the sort of item that makes the case of why replocks on progtot gear are still one of the best things about that system.

MrMelbert commented on code/game/objects/items/knives.dm:
```diff
@@ -274,3 +274,33 @@
+	var/thigh_hit = 4 SECONDS
+
+/obj/item/knife/spy/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
```
```suggestion
	. = ..()
	if(!proximity_flag)
		return
```

MrMelbert reviewed (changes requested):
No roundstart. Feel free to put it in tier1-tier2 pool though. (...With balancing)

Pickle-Coding commented:
> > Should...
> 
> 
> 
> I...

Plasmaman envirosuit doesn't really have any significant armour, so it wouldn't really impact the knife stab anyways. I just don't want 100% armour penetration to be passed around so easily.

I think we can imply the armour being airtight by having complete pressure resistance while all body parts are covered. It makes sense for the knife to slip between a gap through riot armour, but it doesn't make sense for it to cut through deathsquad armour like butter. 100% armour penetration should be for something crazy like a singularity sword that is infinitely sharp.

ATH1909 commented:
> backstabs in ss13 are stupid it is not a first person shooter, you have 360 vision and tg does not have any facing/strafing mechanics people run front and back turning 180 degrees in combat all the time

I think you could probably proc it in some ambushes or after a stun, and even if it's impractical, it's still a funny hehe reference

Deadgebert commented:
don't make it an antag locked item, that's cringe

Loafin34 commented:
> Also I don't see any way how someone would figure out how to use this in game.

its a cool knife a spy would spawn with, at worst it's a reliable AP weapon that can wound and at best you can trickstab

Xackii commented:
> No roundstart. Feel free to put it in tier1-tier2 pool though. (...With balancing)

so that only 1 in 20 spies have a knife

Xackii committed:
Apply suggestions from code review

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

MercuryArrow commented:
> 100% armour penetration should be for something crazy like a singularity sword that is infinitely sharp.

It's worth pointing out that most weapons in the codebase currently with 100% armour penetration are *the complete opposite of this* - they're nearly all incredibly mundane blunt objects with 5-10 damage like brooms and costume staffs.

As far as the knife goes, as much as I've wanted a backstab style item for a long time, I do have to agree that this is very overtuned for even a green/yellow bounty reward much less a freebie at roundstart. It would benefit from being put relatively high in the pool and having the numbers toned down a smidgeon.

jughu commented:
HOLD ON CHAT LETS COOK LETS INCLUDE THE EVERYONE DOESNT KNOW
EXPIRIMENTOR FACTOR! THE DEVICE THAT CAN GET ANY KNIFE IN THE GAME AS NON ANTAG ARE WE SURE WE WANT TO GIVE CREW A 30 BRUTE KNIFE

Xackii commented:
> HOLD ON CHAT LETS COOK LETS INCLUDE THE EVERYONE DOESNT KNOW EXPIRIMENTOR FACTOR! THE DEVICE THAT CAN GET ANY KNIFE IN THE GAME AS NON ANTAG ARE WE SURE WE WANT TO GIVE CREW A 30 BRUTE KNIFE


it only deals 10 damage. Only spies receive other bonuses

jughu commented:
so its a item you can test antags with?

Xackii commented:
> so its a item you can test antags with?

yes

necromanceranne commented:
> > so its a item you can test antags with?
> 
> yes

We should avoid adding these whenever possible, or at least provide the means to obfuscate that it is working as intended for you. That said, we do have rules in place to handle this. Nobody can or should compel you to take an antag test, and you can ahelp anyone who does.

SomeoneYouProbablyKnow commented:
Honestly antag testing concerns aside it's just not an interesting item. It's a finicky damage stick with a few extra bits and bobs you have to keep track of. Why should i be more excited about this being on the bounty board than like an esword or fists of the north star or some other basic weapon?

The backstab is also eh, sometimes it's a bit more damage but you can only proc it reliably when they see you but don't expect you to attack. So in a conversation, so fuck that.

Also hate anything that needs you to crawl on the floor but that's just me.

Xackii commented:
> Honestly antag testing concerns aside it's just not an interesting item. It's a finicky damage stick with a few extra bits and bobs you have to keep track of. Why should i be more excited about this being on the bounty board than like an esword or fists of the north star or some other basic weapon?
> 
> The backstab is also eh, sometimes it's a bit more damage but you can only proc it reliably when they see you but don't expect you to attack. So in a conversation, so fuck that.
> 
> Also hate anything that needs you to crawl on the floor but that's just me.

The logic is rather that if someone forces you, then you will have a chance to fight back. 
I also think that this is too weak an item to be a reward for a spy. The point is to add something unique to the spy + haha ​​funny backstab

Deadgebert commented:
> > Honestly antag testing concerns aside it's just not an interesting item. It's a finicky damage stick with a few extra bits and bobs you have to keep track of. Why should i be more excited about this being on the bounty board than like an esword or fists of the north star or some other basic weapon?
> > The backstab is also eh, sometimes it's a bit more damage but you can only proc it reliably when they see you but don't expect you to attack. So in a conversation, so fuck that.
> > Also hate anything that needs you to crawl on the floor but that's just me.
> 
> The logic is rather that if someone knocks you down, then you will have a chance to fight back. I also think that this is too weak an item to be a reward for a spy. The point is to add something unique to the spy + haha ​​funny backstab

too weak? buddy, this is a 4 hit kill weapon with 100 armor pen that inflicts knockdown, this has better chance to kill someone than an esword. I walk at someone i want to kill, drop to the floor, and start going to town. only thing this is missing is some block chance.
also antag test items are really bad, you'd get more support if you just let it work for everyone.

Sakamoto4ka commented:
> > > Honestly antag testing concerns aside it's just not an interesting item. It's a finicky damage stick with a few extra bits and bobs you have to keep track of. Why should i be more excited about this being on the bounty board than like an esword or fists of the north star or some other basic weapon?
> > > The backstab is also eh, sometimes it's a bit more damage but you can only proc it reliably when they see you but don't expect you to attack. So in a conversation, so fuck that.
> > > Also hate anything that needs you to crawl on the floor but that's just me.
> > 
> > 
> > The logic is rather that if someone knocks you down, then you will have a chance to fight back. I also think that this is too weak an item to be a reward for a spy. The point is to add something unique to the spy + haha ​​funny backstab
> 
> too weak? buddy, this is a 4 hit ~kill~ crit weapon with 100 armor pen that inflicts knockdown, this has better chance to kill someone than an esword. I walk at someone i want to kill, drop to the floor, and start going to town. only thing this is missing is some block chance. also antag test items are really bad, you'd get more support if you just let it work for everyone.

Good luck to make a backstab for 4hit crit

Deadgebert commented:
> Good luck to make a backstab for 4hit crit

https://www.youtube.com/watch?v=mwxcwztbmDE
k

Xackii commented:
> > Good luck to make a backstab for 4hit crit
> 
> https://www.youtube.com/watch?v=mwxcwztbmDE k
> 
> knockdown is one of the strongest things you can add to a weapon right now. even without shove to paralyze. it forces them to drop what they're holding and it slows their movement speed by a very significant amount. This knife has that AND synergy with paralyze shove and extra damage to prone.

Whenever you want to call a weapon strong, compare it to a stun baton. In order for you to pose a real threat to the enemy, you need to keep a few points in mind: I can do more damage if I knock down the enemy, I can knock down the enemy if I lay down myself. In your example, the doll stood motionless. If you are attacked by a sec with a stun baton and a laser, then while you are running towards him to lie down next to him, he will shoot you with a laser. Even a sword is much more useful than a knife. The point is that the spy must provide some kind of threat as an antagonist, and not just run around and steal station items in the hope that he will get a double sword or a bomb.

Sakamoto4ka commented:
> > Good luck to make a backstab for 4hit crit
> 
> https://www.youtube.com/watch?v=mwxcwztbmDE k
> 
> knockdown is one of the strongest things you can add to a weapon right now. even without shove to paralyze. it forces them to drop what they're holding and it slows their movement speed by a very significant amount. This knife has that AND synergy with paralyze shove and extra damage to prone.

now do that on moving spaceman

Loafin34 commented:
this occupies the same niche as rebar crossbow imo. it's powerful for a short burst, but you're not going to fool anyone by laying down with a knife in hand after they've confirmed it's the calf-slasher. also maybe add the telebaton delay thing so people can't thighstun for a really long period of time.

Xackii commented:
> this occupies the same niche as rebar crossbow imo. it's powerful for a short burst, but you're not going to fool anyone by laying down with a knife in hand after they've confirmed it's the calf-slasher. also maybe add the telebaton delay thing so people can't thighstun for a really long period of time. EDIT: it looks like you can't use thighstab for 4 seconds after doing it, so good foresight


Yes, as you have already read, a blow to the thigh will not work if the person is already lying down at that moment (and a blow to a person lying down will not give a knockdown)

Xackii committed:
notroundstart

Xackii commented:
Made knife not a round start, but a reward. Added notifications for the enemy when he is hit in the back or thigh. Added a funny phrase when knife crit the enemy.

Xackii committed:
Merge branch 'master' into spyknife

MrMelbert commented on code/game/objects/items/knives.dm:
```diff
@@ -274,3 +274,45 @@
+		if(((user.dir == living_target.dir) || (HAS_TRAIT(living_target, TRAIT_SPINNING))) && (living_user.body_position != LYING_DOWN))
+			living_target.apply_damage(backstab_bonus, BRUTE)
+			to_chat(living_target, span_danger("you feel severe pain in your back!"))
+		if((living_target.stat == SOFT_CRIT) && epic_say_ready)
```
This looks like it would get rather annoying so let's restrict it to April Fools like some other meme force-says we have in the codebase
```suggestion
		if((living_target.stat == SOFT_CRIT) && check_holidays(APRIL_FOOLS) && epic_say_ready)
```

MrMelbert commented on code/game/objects/items/knives.dm:
```diff
@@ -274,3 +274,45 @@
+		if((living_target.stat == SOFT_CRIT) && epic_say_ready)
+			living_user.say("You got blood on my suit!", forced = "spy knife backstab")
+			epic_say_ready = FALSE
+			addtimer(CALLBACK(src, PROC_REF(epic_say_ready)), 5 SECONDS)
```
You can use `VARSET_CALLBACK` for this, slightly easier

MrMelbert commented on code/game/objects/items/knives.dm:
```diff
@@ -274,3 +274,45 @@
+			living_target.apply_damage(backstab_bonus/2, BRUTE)
+		if((living_user.body_position == LYING_DOWN) && (living_target.body_position != LYING_DOWN))
+			living_target.Knockdown(thigh_hit)
+			to_chat(living_target, span_danger("you feel like something sharp has wounded your thigh!"))
```
```suggestion
			to_chat(living_target, span_userdanger("You feel a sharp pain in your thigh!"))
```

MrMelbert replied:
But also: Consider what would happen if someone with no legs was stabbed, then this chat message would make less sense

MrMelbert commented on code/game/objects/items/knives.dm:
```diff
@@ -274,3 +274,45 @@
+	if((isliving(target)) && (isliving(user)))
+		var/mob/living/living_target = target
+		var/mob/living/living_user = user
+		if((living_target.body_position == LYING_DOWN))
+			living_target.apply_damage(backstab_bonus/2, BRUTE)
+		if((living_user.body_position == LYING_DOWN) && (living_target.body_position != LYING_DOWN))
+			living_target.Knockdown(thigh_hit)
+			to_chat(living_target, span_danger("you feel like something sharp has wounded your thigh!"))
+		if(((user.dir == living_target.dir) || (HAS_TRAIT(living_target, TRAIT_SPINNING))) && (living_user.body_position != LYING_DOWN))
+			living_target.apply_damage(backstab_bonus, BRUTE)
+			to_chat(living_target, span_danger("you feel severe pain in your back!"))
```
We can format this whole block to be a lot easier on the eyes: 
```suggestion
		if(living_target.body_position == LYING_DOWN)
			living_target.apply_damage(backstab_bonus/2, BRUTE)
		else if(living_user.body_position == LYING_DOWN)
			living_target.Knockdown(thigh_hit)
			to_chat(living_target, span_danger("you feel like something sharp has wounded your thigh!"))
		else if(user.dir == living_target.dir || HAS_TRAIT(living_target, TRAIT_SPINNING))
			living_target.apply_damage(backstab_bonus, BRUTE)
			to_chat(living_target, span_danger("you feel severe pain in your back!"))
```

MrMelbert replied:
```suggestion
		if((living_target.body_position == LYING_DOWN))
			living_target.apply_damage(backstab_bonus/2, BRUTE)
		if((living_user.body_position == LYING_DOWN) && (living_target.body_position != LYING_DOWN))
			living_target.Knockdown(thigh_hit)
			to_chat(living_target, span_danger("you feel like something sharp has wounded your thigh!"))
		if(((user.dir == living_target.dir) || (HAS_TRAIT(living_target, TRAIT_SPINNING))) && (living_user.body_position != LYING_DOWN))
			living_target.apply_damage(backstab_bonus, BRUTE)
			to_chat(living_target, span_userdanger("You feel severe pain in your back!"))
```

MrMelbert commented on code/game/objects/items/knives.dm:
```diff
@@ -274,3 +274,45 @@
+		return
+	if(!(user?.mind?.has_antag_datum(/datum/antagonist/spy))) //only a spy can use potential of a knife
+		return
+	if((isliving(target)) && (isliving(user)))
```
```suggestion
	if(isliving(target) && isliving(user))
```

MrMelbert commented on code/game/objects/items/knives.dm:
```diff
@@ -274,3 +274,45 @@
+	inhand_icon_state = "spyknife"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	desc = "Knife for real gentlemen."
```
```suggestion
	desc = "A knife for real gentlemen."
```

MrMelbert commented on code/modules/uplink/uplink_items/spy_unique.dm:
```diff
@@ -11,6 +11,12 @@
 
+/datum/uplink_item/spy_unique/spy_knife
+	name = "Spy Knife"
+	desc = "Knife for real gentlemen."
```
```suggestion
	desc = "A special knife which can deal extra damage while stabbing to the back."
```

MrMelbert commented on code/modules/uplink/uplink_items/spy_unique.dm:
```diff
@@ -11,6 +11,12 @@
+	name = "Spy Knife"
+	desc = "Knife for real gentlemen."
+	item = /obj/item/knife/spy
+	cost = 1
```
Keep it in the `SPY_LOWER_COST_THRESHOLD` range, so it can appear in easy or medium loot pools. 
```suggestion
```

MrMelbert reviewed (changes requested):
Numbers still need some tweaking, still feels a tad overtuned. 

Against a lying down target it's a 25 force weapon with 20 wound bonus and 100 armor pen. 
Compared to an armblade, which is a 25 force weapon with 10 wound bonus and 35 armor pen... this is a little goofy.

Xackii commented:
> Numbers still need some tweaking, still feels a tad overtuned.
> 
> Against a lying down target it's a 25 force weapon with 20 wound bonus and 100 armor pen. Compared to an armblade, which is a 25 force weapon with 10 wound bonus and 35 armor pen... this is a little goofy.

do you mean high damage or high armor penetration (or both)?

Xackii committed:
Apply suggestions from code review

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Xackii committed:
Apply suggestions from code review

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Xackii committed:
redoredoredo

Xackii committed:
legcheck

Xackii committed:
Merge branch 'master' into spyknife

MrMelbert commented:
Both, but keep in mind the bonus damage won't be affected by armor anyways, so you can give it lower AP and it won't matter too much. (maybe 50 ap instead of 100?)

Xackii committed:
pndown

Xackii committed:
Merge branch 'spyknife' of https://github.com/Xackii/tgstation into spyknife

Xackii commented:
> Both, but keep in mind the bonus damage won't be affected by armor anyways, so you can give it lower AP and it won't matter too much. (maybe 50 ap instead of 100?)

did it to 50

Xackii committed:
Merge branch 'master' into spyknife

Wallemations reviewed (approved):
yea sure sprite looks fine

optimumtact commented:
make a new PR when melbert is ready to merge this.
